game/database.cc

```diff
@@ -127,19 +127,30 @@ void Database::queryPerPlayerHiscore(std::ostream & os, std::string const& track
 	}
 }
 
-bool Database::hasHiscore(Song const& s) const 
+bool Database::hasHiscore(Song const& s) const
 try {
 	return m_hiscores.hasHiscore(m_songs.lookup(s).value());
 } catch (const std::exception &e) {
 	std::clog << "database/error: Invalid song ID for song: " + s.artist + " - " + s.title << std::endl;
 	return false;
 }
 
-unsigned Database::getHiscore(const Song& s) const try {
+unsigned Database::getHiscore(Song const& s) const try {
 	const auto songid = m_songs.lookup(s);
 	return m_hiscores.getHiscore(songid.value());
 } catch (const std::exception& e) {
 	std::clog << "database/error: Invalid song ID for song: " + s.artist + " - " + s.title << std::endl;
 	return 0;
 }
 
+unsigned Database::getHiscore(SongPtr const& s) const {
```

Comment:
Idon't really see why we would provide an api using a pointer here, then why not weak_ptr, unique_ptr, raw_ptr, and why not optional....etc

Reply:
It is a const reference and the consumer of this function has a SongPtr. The why not working with this pointer?

Reply:
one could just say the same thing about any function you can write. The point here is that there is no good reason to takt a pointer here, so this is not the responsability of getHiscore to manage upper layer use of songs (that store songs as pointers)

Reply:
I'm gonna resolve this so we can merge. @OznOg maybe explain this better when you come back and then decide whether we want to refactor it in another PR?